src/kboot.c

```diff
@@ -2383,6 +2384,37 @@ static int dt_transfer_virtios(void)
     return 0;
 }
 
+static int dt_set_pmgr(void)
+{
+    if (chip_id != T8012)
+        return 0;
+
+    if (mem_size_actual > 0x40000000)
+        return 0;
+
+    int pmgr_node = fdt_path_offset(dt, "/soc/power-management@20e000000");
+    if (pmgr_node < 0) {
+        printf("FDT: Failed to find pmgr node\n");
+        return 0;
+    }
+
+    int dcs2_node = fdt_path_offset(dt, "/soc/power-management@20e000000/power-controller@80258");
+    if (dcs2_node < 0)
+        bail("FDT: failed to find ps_dcs2 node\n");
+
+    /* Allow failure */
+    fdt_delprop(dt, dcs2_node, "apple,always-on");
```

Comment:
Any reason this isn't just done in the devicetree?

Reply:
It can't be done in the device tree because the amount of memory and the amount of used memory channels depends on the storage configuration of the host Mac.

Reply:
Oh, I totally missed the mem_size_actual check. I'd add a comment to this function then to explain what's going on here.